### DISCUSSION TITLE
Question: (Best Practices) Where do you place your transloco structrural directives in the HTML templates?

### DISCUSSION THREAD

**Discussion by eulersson** [ngneat/transloco#480] [Q&A]
Hello, I have a question about the structural directive. Where should I put it? Should I bubble it up to the uppermost HTML DOM element that's the common ancestor for all the places where there are translations and calls to `t()`? Or should I simply add it blindly in each component that uses translations in a <ng-container *transloco="let t;">`...?

What's the best practice? Do you place them? Do you enclose your component HTMLs with an `<ng-container *transloco="let t;>`? Do you have multiple of those directives within the same HTML template?

**Comment by pavliy** [comment]
As for me, I would go with such mindset:
- If component has root element - it's absolutely OK to put directive there. Don't see any sense in adding one more ng-container just for fun
- If component HAS NO root element and I need to translate whole component - then ng-container is a good deal
- Finally, sometimes if there is component where part is translated and part is not - I guess it's OK to wrap just translatable part - just to indicate 'hey, we do translate here'. But for most of cases I'd go with point 1 or point 2..

**Comment by eulersson** [comment]
Nice one Eugene! I like it. Thanks for sharing it.
